cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -548,6 +560,32 @@ private static void validateConfiguration(Reconciliation reconciliation, Kafka k
         }
     }
 
+    /**
+     * Validates the user configuration with the configuration of quotas plugin
+     * In case that user configured the client.quota.callback.class option and the {@link QuotasPluginStrimzi} is configured as well,
+     * the warning is raised and the option is removed
+     *
+     * @param configuration     {@link KafkaConfiguration} with user specified config
+     * @param quotasPlugin      configuration of the quotas plugin
+     * @param warnings          list of warnings
+     */
+    private static void validateConfigurationOfQuotasPlugin(KafkaConfiguration configuration, QuotasPlugin quotasPlugin, List<Condition> warnings) {
+        final String clientCallbackClassOption = "client.quota.callback.class";
+
+        if (quotasPlugin instanceof QuotasPluginStrimzi quotasPluginStrimzi) {
+            if (quotasPluginStrimzi.getMinAvailableBytesPerVolume() != null && quotasPluginStrimzi.getMinAvailableRatioPerVolume() != null) {
+                throw new InvalidResourceException("You cannot configure both `minAvailableBytesPerVolume` and `minAvailableRatioPerVolume`, they are mutually exclusive.");
+            }
+        }
+
+        if (quotasPlugin != null && configuration.getConfigOption(clientCallbackClassOption) != null) {
+            warnings.add(StatusUtils.buildWarningCondition("QuotasPluginConflict",
+                String.format("Quotas plugin class cannot be configured in .spec.kafka.config, when .spec.kafka.quotas contains configuration of `%s` plugin", QuotasPluginStrimzi.TYPE_STRIMZI)));
```

Comment:
How do you know it is _type Strimzi_?
* The warning here will be likely triggered even when the `type: kafka` configuration is used, or?
* I would argue that it should be triggered since the presence of the configuration suggests that there will be conflict.

Reply:
I will add check that it's type of `QuotasPluginStrimzi`, thanks